File: API/API/repositorys/IUserRepository.vb
Public Interface IUserRepository

    Function GetByEmail(email As String) As User

    Function GetById(id As Guid) As User

    Function Exists(email As String) As Boolean

    Sub Create(user As User)

    Sub Save()

    Function SearchUsers(keyword As String) As List(Of UserSearchResponse)

End Interface

File: API_v2/Repositorys/IRepositorys/IUserRepository.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using API_v2.Models;
using API_v2.Models.DTOs;

namespace API_v2.Repositorys.IRepositorys
{
    public interface IUserRepository
    {
        Task<User?> GetByEmailAsync(string email);
        Task<User?> GetByIdAsync(Guid id);
        void Create(User user);
        Task SaveAsync();
        Task<List<UserSearchResponse>> SearchUsersAsync(string keyword);
    }
}
